packages/web3/src/signer/tx-builder.ts

```diff
@@ -62,28 +68,126 @@ export abstract class TransactionBuilder {
     params: SignTransferTxParams,
     publicKey: string
   ): Promise<Omit<SignTransferTxResult, 'signature'>> {
+    const data = this.buildTransferTxParams(params, publicKey)
+    const response = await this.nodeProvider.transactions.postTransactionsBuild(data)
+    return { ...response, gasPrice: fromApiNumber256(response.gasPrice) }
+  }
+
+  async buildDeployContractTx(
+    params: SignDeployContractTxParams,
+    publicKey: string
+  ): Promise<Omit<SignDeployContractTxResult, 'signature'>> {
+    const data = this.buildDeployContractTxParams(params, publicKey)
+    const response = await this.nodeProvider.contracts.postContractsUnsignedTxDeployContract(data)
+    const contractId = binToHex(contractIdFromAddress(response.contractAddress))
+    return { ...response, groupIndex: response.fromGroup, contractId, gasPrice: fromApiNumber256(response.gasPrice) }
+  }
+
+  async buildExecuteScriptTx(
+    params: SignExecuteScriptTxParams,
+    publicKey: string
+  ): Promise<Omit<SignExecuteScriptTxResult, 'signature'>> {
+    const data = this.buildExecuteScriptTxParams(params, publicKey)
+    const response = await this.nodeProvider.contracts.postContractsUnsignedTxExecuteScript(data)
+    return { ...response, groupIndex: response.fromGroup, gasPrice: fromApiNumber256(response.gasPrice) }
+  }
+
+  async buildChainedTx(params: BuildChainedTxParams[]): Promise<BuildChainedTxResult[]> {
+    const data: BuildTransaction[] = params.map((param) => {
+      switch (param.type) {
+        case 'Transfer': {
+          const value = this.buildTransferTxParams(param, param.publicKey)
+          return { type: param.type, value }
+        }
+        case 'DeployContract': {
+          const value = this.buildDeployContractTxParams(param, param.publicKey)
+          return { type: param.type, value }
+        }
+        case 'ExecuteScript': {
+          const value = this.buildExecuteScriptTxParams(param, param.publicKey)
+          return { type: param.type, value }
+        }
+        default:
+          throw new Error(`Unsupported transaction type: ${(param as any).type}`)
```

Comment:
`(param as any).type` is a bit suspicious as we do `switch (param.type)` a couple of lines above and bellow `${buildResult.type}`

Reply:
Seems that typescript doesn't allow us to do `param.type` since it is already exhaustive (type-wise) for the switch statement, so it thinks it will not end up here.

Reply:
Will get the value before switch statement to get rid of `as any`, thanks :+1: